backend/src/main/java/com/huellapositiva/domain/service/RemoteStorageService.java

```diff
@@ -0,0 +1,29 @@
+package com.huellapositiva.domain.service;
+
+import com.huellapositiva.infrastructure.StorageService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+
+@Service
+@RequiredArgsConstructor
+public class RemoteStorageService {
+
+    @Autowired
+    private final StorageService storageService;
+
+    public URL uploadProposalImage(MultipartFile file, String proposalId) throws IOException {
+        String destinationFileName = proposalId + "." + file.getContentType();
```

Comment:
Since all the images will be publicly available from S3, please make sure the name/route of the images is not guessable otherwise anyone will be able access all of them.